static-metric/examples/make_auto_flush_static_metric.rs

```diff
@@ -0,0 +1,413 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+/*!
+
+Use metric enums to reuse possible values of a label.
+
+*/
+#[macro_use]
+extern crate lazy_static;
+extern crate coarsetime;
+extern crate prometheus;
+extern crate prometheus_static_metric;
+
+use prometheus::*;
+use prometheus_static_metric::auto_flush_from;
+use prometheus_static_metric::make_auto_flush_static_metric;
+
+make_auto_flush_static_metric! {
+
+
+    pub label_enum FooBar {
+        foo,
+        bar,
+    }
+
+    pub label_enum Methods {
+        post,
+        get,
+        put,
+        delete,
+    }
+
+    pub struct Lhrs: LocalIntCounter {
+        "product" => FooBar,
+        "method" => Methods,
+        "version" => {
+            http1: "HTTP/1",
+            http2: "HTTP/2",
+        },
+    }
+}
+
+lazy_static! {
+pub static ref HTTP_COUNTER_VEC: IntCounterVec =
+register_int_counter_vec ! (
+"http_requests",
+"Total number of HTTP requests.",
+& ["product", "method", "version"]    // it doesn't matter for the label order
+).unwrap();
+}
+
+// Macro expanded code of auto_flush_from!
+// lazy_static! {
+//     pub static ref TLS_HTTP_COUNTER: Lhrs = {
+//         thread_local! {
+//             pub static TLS_HTTP_COUNTER_INNER: LhrsInner = LhrsInner::from(& HTTP_COUNTER_VEC);
+//         }
+//         Lhrs::from(&TLS_HTTP_COUNTER_INNER)
+//     };
+// }
+
+auto_flush_from! {
+    TLS_HTTP_COUNTER : Lhrs = HTTP_COUNTER_VEC
```

Comment:
I think this style does not seem very idiomatic...

Reply:
And it seems to hide too many details. I think we should still leave `lazy_static` to users, like:

```rust
lazy_static! {
     pub static ref TLS_HTTP_COUNTER: Lhrs = auto_flush_from!(HTTP_COUNTER_VEC, Lhrs)
}
```

But I am also not quite satisfied with the design above.. Do you have any better ideas?